src/test/java/com/wikia/webdriver/common/core/elemnt/Wait.java

```diff
@@ -315,11 +314,43 @@ public boolean forTextInElement(List<WebElement> elements, int index, String tex
     }
   }
 
+  /**
+   * Wait for successful (http response code less than 400) response from specific service
+   *
+   * @param url Url which was used for making request
+   */
+  public void forSuccessfulResponse(final NetworkTrafficInterceptor networkTrafficInterceptor,
+                                    final String url) {
+    changeImplicitWait(0, TimeUnit.SECONDS);
+
+    try {
+
+      wait.until(
+          new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver webDriver) {
+              return networkTrafficInterceptor.searchRequestUrlInHar(url) &&
+                     networkTrafficInterceptor.getResponseHttpStatusCode(url) < 400;
```

Comment:
First of all: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
Redirection I am considering as a successful response.
And actually somebody already used `< 400` condition for determining whether a http response is successful:
https://github.com/Wikia/selenium-tests/blob/master/src/test/java/com/wikia/webdriver/common/core/networktrafficinterceptor/NetworkTrafficInterceptor.java#L60